### PR TITLE
Workspace config

### LINKED ISSUE #2: Setup repository workspace
**Task description**
Set up repository with workspace for applications needed to facilitate governance committee elections.

**Sub-tasks**
- [x] Add yarn workspace for frontend apps
- [x] Configure eslint + prettier + tsconfig base
- [x] Setup CI

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Sets up yarn workspace to be used by:
- Voting dApp
- Guarding application

Closes #2 

## Changes

- Added yarn workspaces root config files
- Setup lint/formatter (+ pre commit hook)
- Setup voting app project

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (abizjak) [inline .husky/pre-commit:4]: I really don't like pre-commit hooks that depend on anything that is not completely standard. Can we please not have them?

I don't care if you set it up for yourself, but I don't want it in the repository for myself.

File: apps/voting/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>Vite + React + TS</title>
```
Comment (abizjak) [inline apps/voting/index.html:7]: We could probably change this already.

File: .eslintrc.cjs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+module.exports = {
```
Comment (abizjak) [inline .eslintrc.cjs:1]: Why does this have to be polluting the root of the repository?

We'll have a number of non-JS tools tools as well. Do we have to have this here?

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,10 @@
+root = true
+
+[*]
+indent_style = space
+max_line_length = 120
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (abizjak) [inline .editorconfig:10]: Since we have a formatter I don't see a need for this.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (abizjak) [inline package.json:1]: Why is this in the root of the repository?

We'll have a backend, and some other Rust tools in here. I find it annoying to pollute the root with files like this.

Wouldn't it make more sense to put all the JS stuff under some frontend directory.

**Review by abizjak — COMMENTED** [review]
We should set up CI for the voting project as part of this PR as well.

File: .editorconfig (hunk 4 above)
Comment (soerenbf) [reply .editorconfig:10]: I'll remove it and leave it to the formatters.

File: .eslintrc.cjs (hunk 3 above)
Comment (soerenbf) [reply .eslintrc.cjs:1]: Will move to frontend apps workspace

File: .husky/pre-commit (hunk 1 above)
Comment (soerenbf) [reply .husky/pre-commit:4]: I'll remove this and set up CI

File: .github/workflows/check.yml [hunk 6]
```diff
@@ -0,0 +1,103 @@
+name: Check
```
Comment (abizjak) [inline .github/workflows/check.yml:1]: Can you please add a comment here at the top stating what this is checking

File: .github/workflows/check.yml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      - name: Restore dependencies
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ./apps/node_modules
+            ./apps/voting/node_modules
+          key: ${{ runner.os }}-yarn
+
+      - name: Lint
+        run: yarn build:all
```
Comment (abizjak) [inline .github/workflows/check.yml:103]: Shouldn't we have a step which does type-checking? Does `lint` do type-checking?

**Review by abizjak — APPROVED** [review]

File: .github/workflows/check.yml (hunk 7 above)
Comment (soerenbf) [reply .github/workflows/check.yml:103]: Build does type checks

File: .github/workflows/check.yml (hunk 6 above)
Comment (soerenbf) [reply .github/workflows/check.yml:1]: Will do 🙂

**Pull request merged**